linkml/generators/javagen.py

```diff
@@ -42,9 +42,9 @@
 
 TYPEMAP = {
     "str": "String",
-    "int": "Integer",
-    "float": "Float",
-    "Bool": "Boolean",
+    "int": "int",
+    "float": "float",
```

Comment:
it looks like you are mapping from python types - I think you should map from the uri of the type or the name of the type, this gives you the distinction eg between float and double. See #777

Reply:
Here is the problem:

https://github.com/linkml/linkml/blob/29a64763d113a7ee6305d73765c432661858241c/linkml/generators/javagen.py#L70-L71

We map from t.base, which has already lost information.

```yaml
  float:
    uri: xsd:float
    base: float
    description: A real number that conforms to the xsd:float specification

  double:
    uri: xsd:double
    base: float
    description: A real number that conforms to the xsd:double specification
```

instead map from t.uri

I also recommend using sv.induced_type which should propagate uris from parent types

Reply:
See also https://github.com/linkml/linkml/issues/794